fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/RecordMetaDataBuilderTest.java

```diff
@@ -306,4 +306,21 @@ public void noPrimaryKey() {
                 RecordMetaData.newBuilder().setRecords(TestRecordsNoPrimaryKeyProto.getDescriptor()).getRecordMetaData());
         assertEquals("Record type MyNoPrimaryKeyRecord must have a primary key", t.getMessage());
     }
+
+    @Test
+    public void updateRecords() {
+        RecordMetaDataBuilder builder = RecordMetaData.newBuilder().setRecords(TestRecords1Proto.getDescriptor());
+        assertNotNull(builder.getRecordType("MySimpleRecord"));
+        assertTrue(builder.getRecordType("MySimpleRecord").getDescriptor().getFile() == TestRecords1Proto.getDescriptor());
```

Comment:
`assertSame(a, b)` instead of `assertTrue(a == b)` here and elsewhere. They amount to the same thing, but the error messages are much more understandable using `assertSame`.